src/kessel/auth/index.ts

```diff
@@ -1,31 +1,72 @@
 import type * as oauth from "oauth4webapi";
 import { ClientConfigAuth } from "../inventory";
 
-const EXPIRATION_WINDOW = 20000; // 20 seconds in milliseconds
+const EXPIRATION_WINDOW_MILLI = 300000; // 5 minutes in milliseconds
+const DEFAULT_EXPIRE_IN_SECONDS = 3600; // 1 hour in seconds
 
 interface TokenData {
   accessToken: string;
   expiresAt: number;
 }
 
+interface RefreshTokenResponse {
+  accessToken: string;
+  expiresAt: number;
```

Comment:
To follow closer to the spec, can we make this use the `Date` object? We're already calling Date.now() to the get current time, could probably wrap the final value in a date object as well instead of just an integer.

Reply:
Though about this, but wasn't convinced since the Date object doesn't really support comparison out of the box. I'll try changing to date and see what's the impact

Reply:
Updated to use `Date` - while there isn't any compare function, there is a function to return the epoch to make it easier to compare.